sys/riscv/riscv/trap.c

```diff
@@ -138,14 +137,17 @@ cpu_fetch_syscall_args(struct thread *td)
 		if (SV_PROC_FLAG(td->td_proc, SV_CHERI))
 			stack_args = (char * __capability)td->td_frame->tf_sp;
 #endif
+	} else {
+		shift_args = false;
```

Comment:
You can avoid a branch by copying the first value here and incrementing the destination pointer and then copying NARGS - 1 below:

https://cheri-compiler-explorer.cl.cam.ac.uk/z/T4fdEa

Reply:
Nice, thanks!